File: scripts/728_Self Dividing Numbers.py
class Solution:
    def isDividingNumber(self, num):
        if '0' in str(num):
            return False
        return 0 == sum(num % int(i) for i in str(num))

    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        answer = []
        for num in range(left, right + 1):
            if self.isDividingNumber(num):
                answer.append(num)
        return answer

if __name__ == "__main__":
    print(Solution().selfDividingNumbers(1, 22))

File: scripts/728_Self Dividing Numbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
function isDividingNumber(num) {
    var digit = num.toString();

    for (var i = 0; i < digit.length; i++) {
        if (digit[i] == 0) {
            return false;
        } else if (num % digit[i] != 0) {
            return false;
        }
    }
    return true;
}

var selfDividingNumbers = function (left, right) {
    var answer = [];
    for (var i = left; i <= right; i++) {
        if (isDividingNumber(i)) {
            answer.push(i);
        }
    }
    return answer;
}; 